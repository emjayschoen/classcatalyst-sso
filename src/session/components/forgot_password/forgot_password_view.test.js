import React from 'react';
import { shallow } from 'enzyme/build';
import ForgotPasswordView from './forgot_password_view';

describe('<ForgotPasswordView />', () => {
  let subject;

  describe('not logged in', () => {
    beforeEach(() => {
      subject = shallow(<ForgotPasswordView signedIn={false} handleSubmit={jest.fn()} />);
    });

    it('renders correctly', () => {
      expect(subject).toMatchSnapshot();
    });
  });

  describe('when already logged in', () => {
    beforeEach(() => {
      subject = shallow(<ForgotPasswordView signedIn handleSubmit={jest.fn()} />);
    });

    it('renders correctly', () => {
      expect(subject).toMatchSnapshot();
    });
  });
});
