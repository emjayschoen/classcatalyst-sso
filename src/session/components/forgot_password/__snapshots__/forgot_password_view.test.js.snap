// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForgotPasswordView /> not logged in renders correctly 1`] = `
<styled.div>
  <form
    onSubmit={[Function]}
  >
    <h3>
      Forgot your password?
    </h3>
    <Field
      component={[Function]}
      fullWidth={true}
      label="Email"
      name="email"
      required={true}
      type="text"
    />
    <input
      className="btn"
      disabled={false}
      type="submit"
      value="Request Password Reset"
    />
    <span
      className="status"
    />
    <p>
      <Link
        to="/signin"
      >
        Sign in
      </Link>
      <span
        style={
          Object {
            "padding": "0 1em",
          }
        }
      >
        |
      </span>
      <Link
        to="/register"
      >
        Don't have an account?
      </Link>
    </p>
  </form>
</styled.div>
`;

exports[`<ForgotPasswordView /> when already logged in renders correctly 1`] = `
<Redirect
  to="/profile"
/>
`;
